### PR TITLE
Support multi trial on gpu

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: src/nni_manager/rest_server/restValidationSchemas.ts [hunk 1]
```diff
@@ -31,10 +31,12 @@ export namespace ValidationSchemas {
                 passwd: joi.string(),
                 sshKeyPath: joi.string(),
                 passphrase: joi.string(),
-                gpuIndices: joi.string()
+                gpuIndices: joi.string(),
+                maxTrialNumOnEachGPU: joi.number()
```
Comment (chicm-ms) [inline src/nni_manager/rest_server/restValidationSchemas.ts]: suggest change config name to: maxTrialNumPerGpu

File: src/nni_manager/training_service/local/gpuScheduler.ts [hunk 2]
```diff
@@ -71,14 +71,15 @@ class GPUScheduler {
...
+    public getAvailableGPUIndices(occupiedGpuIndexNumMap: Map<number, number>): number[] {
         if (this.gpuSummary !== undefined) {
             if(process.platform === 'win32') {
                 return this.gpuSummary.gpuInfos.map((info: GPUInfo) => info.index);
             }
             else{
-                return this.gpuSummary.gpuInfos.filter((info: GPUInfo) => info.activeProcessNum === 0)
-                    .map((info: GPUInfo) => info.index);
+                return this.gpuSummary.gpuInfos.filter((info: GPUInfo) => 
+                occupiedGpuIndexNumMap.get(info.index) === undefined && info.activeProcessNum === 0 ||
```
Comment (QuanluZhang) [inline src/nni_manager/training_service/local/gpuScheduler.ts]: let's reconsider this logic

File: src/nni_manager/training_service/remote_machine/gpuScheduler.ts [hunk 3]
```diff
@@ -93,6 +95,16 @@ export class GPUScheduler {
             rmMeta.gpuReservation.forEach((reserveTrialJobId : string, gpuIndex : number) => {
                 if (reserveTrialJobId === trialJobId) {
                     rmMeta.gpuReservation.delete(gpuIndex);
+                    let occupiedTrialNumPerGPU = this.machineOccupiedTrialNumPerGPUMap.get(rmMeta);
+                    if(occupiedTrialNumPerGPU === undefined) {
+                        throw new Error(`gpu scheduler error!`);
```
Comment (xuehui1991) [inline src/nni_manager/training_service/remote_machine/gpuScheduler.ts]: I am not sure the exception contains enough information to debug...

File: src/nni_manager/rest_server/restValidationSchemas.ts (hunk 1 above)
Comment (SparkSnail) [reply src/nni_manager/rest_server/restValidationSchemas.ts]: changed to use maxTrialNumPerGpu.

File: src/nni_manager/training_service/local/gpuScheduler.ts (hunk 2 above)
Comment (chicm) [reply src/nni_manager/training_service/local/gpuScheduler.ts]: Maybe we should change the linux behaviour here to same as Windows. Let's discuss it.

File: src/nni_manager/training_service/remote_machine/gpuScheduler.ts (hunk 3 above)
Comment (SparkSnail) [reply src/nni_manager/training_service/remote_machine/gpuScheduler.ts]: fixed.

**Pull request closed without merging**